### DISCUSSION TITLE
第3节课作业(7班)

### DISCUSSION THREAD

**Discussion by JimmyMa99** [InternLM/Tutorial#197] [作业]
第3节课作业(7班)

**Comment by createup2023** [comment]
https://www.yuque.com/creatup/ynrsde/aidnmhkg1xwnzgqi?singleDoc# 《基础作业：》
小结：不知道是7b的模型太小还是数据库的问题，显示出的结果没有想象的好。

**Comment by wuhaostudio** [comment]
<img width="1067" alt="微信图片_20240110112807" src="https://github.com/InternLM/tutorial/assets/143194075/78104885-c03d-4e1e-a1c1-a9153ead4ed1">

**Comment by hui1feng** [comment]
第三节课作业：https://github.com/hui1feng/InternLM-LangChain/blob/main/%E4%BD%9C%E4%B8%9A3.ipynb

**Comment by PommesPeter** [comment]
## 第三节课作业

让 InternLM 告诉我 XTuner 的环境配置方法：
![image](https://github.com/InternLM/tutorial/assets/54879512/78d55ad2-ddfc-42bc-90b3-82500754fe7f)

**Comment by lifelinezhang** [comment]
https://zhuanlan.zhihu.com/p/676776368

**Comment by SaaRaaS-1300** [comment]
第三节课，笔记同作业，github：https://github.com/SaaRaaS-1300/InternLM_openNotebook/blob/main/Lesson-3/Lesson-3-Notebook.md

**Comment by snowhou** [comment]
第三课作业（同笔记）：https://zhuanlan.zhihu.com/p/677550010

**Comment by leochiang** [comment]
第三节课作业
<img width="593" alt="20240113111208" src="https://github.com/InternLM/tutorial/assets/7628117/24592216-8c50-43fb-9331-bda30b5eccb8">
第三节课笔记
https://vvgyb242x83.feishu.cn/docx/O3OvdYiwXo6hixxxTeJcyCSNnlb?from=from_copylink

**Comment by abelezm** [comment]
第三课：基于 InternLM 和 LangChain 搭建你的知识库
笔记同作业，https://blog.csdn.net/weixin_38043453/article/details/135575407

**Comment by thefengl** [comment]
# 基于InternLM和LangChain搭建你的知识库

## 基础作业

### 复现课程知识库助手搭建过程 (截图)

1. 下载开源词向量模型Sentence Transformer
![知识库_1](https://github.com/InternLM/tutorial/assets/83425837/a288ada9-eb62-4b26-a961-df6558d6d97b)



2. 下载NLTK相关资源

![知识库_2](https://github.com/InternLM/tutorial/assets/83425837/36cab713-ce0e-4059-8fda-bb11bf0d896b)


3. 构建向量数据库

![知识库_3](https://github.com/InternLM/tutorial/assets/83425837/6ccd2753-83cc-4d0c-badd-00f504816d51)


4. 启动web demo

![知识库_4](https://github.com/InternLM/tutorial/assets/83425837/e7968829-375a-4cbe-baf2-7902d035077d)


5. 测试问答

![知识库_5](https://github.com/InternLM/tutorial/assets/83425837/4e3a0b13-ed9f-4e29-b9e1-36aa35f70d0b)

### debug

问题1：安装huggingface-hub后执行huggingface-cli命令提示无效

解决方法：重装huggingface-hub

**Comment by Rogue-Trader-zzy** [comment]
第三次作业：https://zhuanlan.zhihu.com/p/677702698

**Comment by anneeeee** [comment]
1. Huggingface下载词向量模型
![image](https://github.com/InternLM/tutorial/assets/16276686/ec753b62-c40d-4bc3-a74a-5862cab4a927)

2. 加载NLTK相关资源包
![image](https://github.com/InternLM/tutorial/assets/16276686/0de0767a-e28b-492a-8b0f-df3a308add25)

3. 构建知识库
![image](https://github.com/InternLM/tutorial/assets/16276686/7151a314-4900-4990-b100-0e86ba0153f2)

4. 启动web服务
![image](https://github.com/InternLM/tutorial/assets/16276686/60c95377-7ed6-4418-8c82-a9c87d248e85)


启动web服务一开始报错：
![image](https://github.com/InternLM/tutorial/assets/16276686/9e5997a9-9aea-4065-9245-84a125bce7c5)
经排查是模型路径填写错误，修改后服务启动成功。
![image](https://github.com/InternLM/tutorial/assets/16276686/2100e540-ca76-488a-b524-2f5352592385)

5. web demo体验：
![image](https://github.com/InternLM/tutorial/assets/16276686/9b49dbbb-1ab0-4e3b-bec0-303054796b1b)
![image](https://github.com/InternLM/tutorial/assets/16276686/6e4adad0-14b4-4be2-951d-6ce27be88c61)

**Comment by xiay6** [comment]
第三节课作业打卡
![1](https://github.com/InternLM/tutorial/assets/154853022/a4fab264-0b8e-4808-b9a5-cc6340094bb6)
![2](https://github.com/InternLM/tutorial/assets/154853022/14e84f4a-229a-46bb-99f2-ba9e0c02ee2c)

**Comment by abelezm** [comment]
第三课笔记和作业打卡：https://blog.csdn.net/weixin_38043453/article/details/135575407

**Comment by codeview126** [comment]
复现课程知识库助手搭建过程 (截图)：
结果：
![2024011_04](https://github.com/InternLM/tutorial/assets/127104394/315203b7-7cba-4562-8006-9657d6e25ccf)
过程：随后立即补充

**Comment by Simon-lyz** [comment]
## InternLM Lesson-3 简单RAG
https://g4w90egc30.feishu.cn/docx/LbIydMCbYoF7YsxI8egcAIznnwe?from=from_copylink
